crates/sui-adapter/src/adapter.rs

```diff
@@ -458,19 +460,15 @@ fn process_successful_execution<
                 // unwrap safe because this event can only be emitted from processing
                 // native call delete_id, which guarantees the type of the id.
                 let id: VersionedID = bcs::from_bytes(&event_bytes).unwrap();
+                deleted_ids.insert(id.object_id().clone().into());
                 let obj_id = id.object_id();
                 // We don't care about IDs that are generated in this same transaction
                 // but only to be deleted.
                 if !newly_generated_ids.contains(obj_id) {
                     match by_value_objects.remove(id.object_id()) {
-                        Some((Owner::ObjectOwner { .. }, _)) => {
-                            // If an object is owned by another object, we are not allowed to directly delete the child
-                            // object because this could lead to a dangling reference of the ownership. Such
-                            // dangling reference can never be dropped. To delete this object, one must either first transfer
-                            // the child object to an account address, or call through Transfer::delete_child_object(),
-                            // which would consume both the child object and the ChildRef ownership reference,
-                            // and emit the DeleteChildObject event. These child objects can be safely deleted.
-                            return Err(SuiError::DeleteObjectOwnedObject);
+                        Some((Owner::ObjectOwner(parent_id), _)) => {
+                            parents_to_delete.insert(parent_id);
+                            state_view.delete_object(obj_id, id.version(), DeleteKind::Normal)
```

Comment:
Not quite follow this change. The reason it was implemented this way is because of line 494 where DeleteChildObject is handled separately (which signals that a child object is deleted together with a child ref, hence it's safe). Not sure if you could achieve the same without it.
I think you could make the ID::delete call a separate internal call to distinguish the case?

Reply:
Ahh I see how the old scheme worked now. Your suggested fix makes sense.